### PR TITLE
NPM: Add MathJax (Updated PR #8262)

### PR DESCRIPTION
**Pull request by fneumann**
This PR adds `mathjax` as NPM dependency.

Usage:

- Page editor: text content
- Forum: thread, post, reply
- Survey: introductory message, final statement
- Test Questions: question text, feedback, hints
- Cloze Question: cloze text
- Kprim Question: answers
- Matching Question: terms, definitions
- SC Question: answers
- MC Question: answers
- Ordering Question: answers

Wrapped By:
- components/ILIAS/UI/resources/js/MathJax/mathjax.js 
if PR [Stramline LatexUsage](https://github.com/ILIAS-eLearning/ILIAS/pull/8256) is merged.

Reasoning:

See Feature Request [Streamline Latex Usage](https://docu.ilias.de/go/wiki/wpage_5614_1357)

MathJax is currently added as third-party software.  It is configured with a URL in the ILIAS administration. That allows ILIAS administrators to adapt their MathJax support, but they need the MathJax CDN or must maintain a local copy themselves. This makes it difficult to compare installations and reproduce errors in the ILIAS test installation. Updates in the case of security problems are the responsibility of local administrators. With MathJax as an npm dependency, this gets much more predictable.

Furthermore, ILIAS components that export HTML content can include MathJax in their exported archives instead of linking it from a CDN.

Maintenance:

MathJax is the de-facto standard for rendering LaTeX on web pages. It is supported by strong partners (IEEE, Elsevier). It is actively developed. In the last month, 24 pull requests have been merged to the source repository.

Links:

NPM: https://www.npmjs.com/package/mathjax
GitHub: https://github.com/mathjax
Documentation: https://www.mathjax.org

### CODE REVIEW

**Comment by dsstrassner** [comment]
@fneumann is this still relevant for ILIAS 11? If so, shouldn't this presented to the JF?

**Comment by fneumann** [comment]
Yes, it is relevant. I nearly finished a new proposal for the necessary changes in the UI framework and plan to discuss this all together at the JF in the context of the "Streamline Latex" FR.

**Comment by dsstrassner** [comment]
okay, thanks for the information. but don't forget, that @lscharmer has provided #9086 for all currently accepted deps and I don't know how many effort is need after merging #9086 ;)

**Comment by dsstrassner** [comment]
I added the JF Label as @fneumann added the FR [Streamline LaTeX usage](https://docu.ilias.de/go/wiki/wpage_5614_1357) to the DC. I don't know if @matthiaskunkel adds the FR to the Agenda on 17/03/2025, but this PR should not be forgotten. Please remove the label if the Feature Request will be discussed later on.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 31 MAR 2025: We highly appreciate this PR and accept this dependency for ILIAS 11.

**Comment by lscharmer** [comment]
Merged with #9086
